src/coprocessor/codec/mysql/duration.rs

```diff
@@ -253,97 +254,206 @@ mod parser {
 
 } /* parser */
 
-/// `Duration` is the type for MySQL `time` type.
-///
-/// It only occupies 24 bytes in memory.
-/// TODO: reduce size of `Duration` to 8 bytes
-#[derive(Debug, Clone, Copy)]
-pub struct Duration {
-    dur: StdDuration,
-    /// Fractional Seconds Precision
-    /// See http://dev.mysql.com/doc/refman/5.7/en/fractional-seconds.html
+bitfield! {
+    #[derive(Clone, Copy)]
+    pub struct Duration(u64);
+    impl Debug;
+    #[inline]
+    bool, get_neg, set_neg: 63;
+    #[inline]
+    bool, get_reserved, set_reserved: 62;
+    #[inline]
+    u32, get_hours, set_hours: 61, 48;
+    #[inline]
+    u32, get_minutes, set_minutes: 47, 40;
+    #[inline]
+    u32, get_secs, set_secs: 39, 32;
+    #[inline]
+    u32, get_micros, set_micros: 31, 8;
+    #[inline]
+    u8, get_fsp, set_fsp: 7, 0;
+}
+
+/// Rounds `micros` with `fsp` and handles the carry.
+fn round(
+    hours: &mut u32,
+    minutes: &mut u32,
+    secs: &mut u32,
+    micros: &mut u32,
     fsp: u8,
-    neg: bool,
+) -> Result<()> {
+    if *micros < 1_000_000 {
+        *micros *= 10;
+    }
+
+    *micros = if fsp == 6 {
```

Comment:
These const `6` the same?

Reply:
Seems like this `if` is redundant?

Reply:
This `if` can save one division and one multiplication in rounding when `fsp == 6`